Polyfills/Canvas/Source/Image.cpp

```diff
@@ -91,22 +94,56 @@ namespace Babylon::Polyfills::Internal
         request.SendAsync().then(m_runtimeScheduler, *m_cancellationSource, [env{info.Env()}, this, request{std::move(request)}](arcana::expected<void, std::exception_ptr> result) {
             if (result.has_error())
             {
-                throw Napi::Error::New(env, result.error());
+                const auto error = Napi::Error::New(env, result.error());
+                if (!m_onerrorHandlerRef.IsEmpty())
+                {
+                    m_onerrorHandlerRef.Call({error.Value()});
+                    return;
+                }
+
+                error.ThrowAsJavaScriptException();
```

Comment:
This is still not quite right. We still should interrupt the function when there is no error handler.

```suggestion
                const auto error = Napi::Error::New(env, result.error());
                if (m_onerrorHandlerRef.IsEmpty())
                {
                    error.ThrowAsJavaScriptException();
                }
                else
                {
                    m_onerrorHandlerRef.Call({error.Value()});
                }
                
                return;
```

Reply:
Maybe also put lines 97-105 in a private helper function.

Reply:
Move this code to a helper function, but am still leaving it as ThrowAsJavaScriptException() rather than a native C++ exception, since that would crash the app since it is unhandled.  I don't think it makes sense to crash the whole app in the event a user provides a bad src string.

Reply:
> am still leaving it as ThrowAsJavaScriptException() rather than a native C++ exception

Yup, I wasn't commenting on that part.